File: app/src/main/java/com/example/ama/questapp/domain/engine/QuestEngineGateway.java
package com.example.ama.questapp.domain.engine;

import com.example.ama.questapp.data.db.model.UserTask;

import io.reactivex.Completable;

public interface QuestEngineGateway {

    Completable onIncrementQuestProgress(UserTask userTask);

    Completable addNewRandomDailyQuests(int maxCount);
}

File: app/src/main/java/com/example/ama/questapp/domain/engine/QuestEngineGateway.kt
package com.example.ama.questapp.domain.engine

import com.example.ama.questapp.data.db.model.UserTask
import io.reactivex.Completable

interface QuestEngineGateway {

    fun onIncrementQuestProgress(userTask: UserTask): Completable

    fun addNewRandomDailyQuests(maxCount: Int): Completable
}